src/App.tsx

```diff
@@ -1,41 +1,306 @@
 import './App.css';
 
-import { useState } from 'react';
+import { useEffect, useState } from 'react';
 
-import viteLogo from '/vite.svg';
-
-import reactLogo from './assets/react.svg';
+import { Game2048 } from './page/Layout';
 
 function App() {
-  const [count, setCount] = useState(0);
+  const initialGrid: number[][] = [
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+  ];
+
+  // 초기 설정
+  const [grid, setGrid] = useState<number[][]>(
+    addNewNumber(addNewNumber(initialGrid)),
+  );
+  const [score, setScore] = useState<number>(0);
+  const [bestScore, setBestScore] = useState<number>(
+    parseInt(localStorage.getItem('bestScore') || '0'),
+  );
```

Comment:
```suggestion
  const [bestScore, setBestScore] = useState(
    parseInt(localStorage.getItem('bestScore') ?? '0'),
  );
```

1. `useState(0)` 만 해도 score 가 number로 자동추론되므로 안 써줘도 됩니다! (밑에 `<boolean>` 이런 것도 마찬가지)
2. 린트 에러는 제가 설정해둔 규칙에 의해서 발생하는 건데, `||` 는 모든 falsy value에 대해 right side를 선택하므로 명확하지 않고, null 인 경우를 잡아주고 싶은 거니 `??` 를 사용하는 게 더 코드를 명확하게 만들어준다고 생각해서 해당 규칙 켜뒀습니다!